### DISCUSSION TITLE
Developer docs?

### DISCUSSION THREAD

**Discussion by anguslees** [siderolabs/talos#4189] [Q&A] [answered]
Are there some talos developer docs somewhere?

I'm trying to build an image for an arm64 "NanoPi K1 Plus" board (basically the same as the other Allwinner arm64 boards, but of course needs a different dtb file) from my intel laptop, and I'm a bit bewildered by all the Makefile targets.

I grep'ed for 'bananapi' and made the obvious copies of everything for my new nanopi board.  Great, now how do I test this out?

I _think_ I want to run my new `make sbc-nanopi_k1plus` target, but that fails because it tries to `docker run ghcr.io/talos-systems/installer:$my_git_dirty_ref` which of course doesn't exist.  I tried `make installer` and that builds the right installer image (afaics), but leaves it in the docker buildx builder, where docker run can't find it.  I tried `make docker-installer DEST=/tmp/x && docker import /tmp/x/installer.tar ghcr.io/talos-systems/installer:$my_git_dirty_ref` which appears to work, but trying the `sbc-nanopi_k1plus` make target again now gives a `exec: "image": executable file not found in $PATH"` error from docker run.  I haven't debugged this further yet, because I assume I'm just on the wrong track at this point. :/

What's the build+try workflow _meant_ to look like?

**Comment by rsmitty** [comment, marked as answer]
Hey @anguslees, all of our docs would be found at https://talos.dev which you've probably already seen, but this issue probably just comes down to our default values in the makefile. Try something like `make installer USERNAME=anguslees REGISTRY=docker.io PUSH=true` to actually build and push your own installer to a repo you have access to. Then, try rerunning your new SBC target with the same vars, `make sbc-nanopi_k1plus USERNAME=anguslees REGISTRY=docker.io`.

If that doesn't work, shout back at us here or in slack. Thanks for posting, this is indeed worthy of better documentation.

**Reply by anguslees** [reply]
Thanks, so I was roughly on the right track but didn't push through.  Will persevere :+1:

As some feedback, all the `@` directives in the Makefile make this harder - I've basically had to remove them from every target I use, just to work out what/where things are going wrong.
